ereolen_support/ereolen_support.helper.inc

```diff
@@ -0,0 +1,255 @@
+  $requestParts = _ereolen_support_get_request_parts('PUT', $data);
+  $result = drupal_http_request($requestParts['jira_path'] . '/user?username=' . $jiraUser->name, $requestParts['options']);
+  _ereolen_support_log_error($result);
+  $jiraUser = json_decode($result->data);
+
+  return $jiraUser;
+}
+
+/**
+ * Create an issue in Jira.
+ *
+ * @param array $form
+ *   The support form.
+ * @param array $form_state
+ *   The state of the support form.
+ * @param \stdClass $jiraUser
+ *   A user in Jira.
+ *
+ * @throws \Exception
+ */
+function _ereolen_support_send_to_jira($form, &$form_state, $jiraUser) {
+  $data = [
+    'fields' => [
+      'project' => [
+        'id' => variable_get('jira_project_id'),
+      ],
+      'issuetype' => [
+        'id' => variable_get('jira_issue_type_id'),
+      ],
+      'summary' => substr(preg_replace('/\s+/', ' ', trim($form_state['values']['ereolen_support_description'])), 0, 20),
+      'description' => $form_state['values']['ereolen_support_description'],
+      'reporter' => [
+        'name' => $jiraUser->emailAddress,
+      ],
+    ],
+  ];
+  foreach ($form as $key => $element) {
+    if(is_array($element)) {
+      // Look for form fields.
+      if(strpos($key, 'ereolen_support') === 0) {
+        // Load config related to that field.
+        $jira_field_id = variable_get($key . '_jira_field');
+        // Make sure the config is set and has proper type.
+        if (!empty($jira_field_id) && is_numeric($jira_field_id)) {
+          // Add custom field to data array.
+          $data['fields']['customfield_' . $jira_field_id] = $form_state['values'][$key];
+          if($element['#type'] == 'select') {
+            // Use Label as output from select lists.
+            $data['fields']['customfield_' . $jira_field_id] = $element['#options'][$form_state['values'][$key]];
+          }
+          else {
+            $data['fields']['customfield_' . $jira_field_id] = $form_state['values'][$key];
+          }
+        }
+      }
+    }
+  }
+
+  $requestParts = _ereolen_support_get_request_parts('POST', $data);
+
+  $result = drupal_http_request($requestParts['jira_path'] . '/issue', $requestParts['options']);
+  _ereolen_support_log_error($result);
+}
+
+/**
+ * Create the parts required to make a drupal_http_request.
+ *
+ * @param string $method_type
+ *   The request method.
+ * @param array $data
+ *   An optional data array.
+ *
+ * @return array
+ *   An array containing base path for the request and the options related to
+ *   the request.
+ *
+ * @return array
+ */
+function _ereolen_support_get_request_parts($method_type, $data = []) {
+  $jiraUsername = variable_get('jira_username');
+  $jiraPassword = variable_get('jira_password');
+  // Jira requires base64_encode auth.
+  $encoded_auth = base64_encode($jiraUsername . ':' . $jiraPassword);
+  $jira_api_path = variable_get('jira_domain') . '/rest/api/latest';
+
+  $options = array(
+    'method' => $method_type,
+    'timeout' => 15,
```

Comment:
Hardcoded timeout?

What is the unit for this timeout?

Reply:
Hmm is that bad practice? Added comment about unit.